migrations/2_deploy.js

```diff
@@ -55,30 +53,73 @@ const TokenA = artifacts.require("TokenA");
 const TokenB = artifacts.require("TokenB");
 const FeeToken = artifacts.require("TokenC");
 
+// External contracts
+let { ZeroExExchangeV1, ZeroExProxyV1 } = require("../test/contracts/ZeroExV1");
+let { ZeroExExchangeV2, ZeroExProxyV2 } = require("../test/contracts/ZeroExV2");
+
 // Other constants
 const BigNumber = require('bignumber.js');
 const ONE_HOUR = new BigNumber(60 * 60);
 
-function maybeDeployTestTokens(deployer, network) {
+// Helper functions
+function parseExternalContracts(contracts) {
+  const classDefaults = {
+    from: web3.eth.accounts[0],
+    gas: 6721975,
+    gasPrice: 100000000000
+  };
+  for (let i in contracts) {
+    let contract = contracts[i];
+    contract.setProvider(web3.currentProvider);
+    contract.setNetwork(web3.currentNetwork);
+    contract.class_defaults = classDefaults;
+  }
+}
+
+// Deploy functions
+async function maybeDeployTestTokens(deployer, network) {
+  if (isDevNetwork(network)) {
+    await Promise.all([
+      deployer.deploy(TokenA),
+      deployer.deploy(TokenB),
+      deployer.deploy(FeeToken)
+    ]);
+  }
+}
+
+async function maybeDeploy0xV1(deployer, network) {
   if (isDevNetwork(network)) {
-    return deployer.deploy(TokenA)
-      .then(() => deployer.deploy(TokenB))
-      .then(() => deployer.deploy(FeeToken));
+    parseExternalContracts([ZeroExExchangeV1, ZeroExProxyV1]);
+    await deployer.deploy(ZeroExProxyV1);
+    await deployer.deploy(ZeroExExchangeV1, FeeToken.address, ZeroExProxyV1.address);
+    const proxy = await ZeroExProxyV1.deployed();
+    await proxy.addAuthorizedAddress(ZeroExExchangeV1.address);
   }
-  return Promise.resolve(true);
 }
 
-function maybeDeploy0x(deployer, network) {
+async function maybeDeploy0xV2(deployer, network) {
   if (isDevNetwork(network)) {
-    return deployer.deploy(ZeroExProxyV1)
-      .then(() => deployer.deploy(ZeroExExchangeV1, FeeToken.address, ZeroExProxyV1.address))
-      .then(() => ZeroExProxyV1.deployed())
-      .then(proxy => proxy.addAuthorizedAddress(ZeroExExchangeV1.address));
+    parseExternalContracts([ZeroExExchangeV2, ZeroExProxyV2]);
+    const zrxAssetData = "0xf47261b0000000000000000000000000" + FeeToken.address.substr(2);
```

Comment:
I think you should not do this kind of stuff manually but should instead use the functions provided by 0x.js. If you need to install multiple versions of 0x.js see https://github.com/dydxprotocol/orderbook/blob/master/package.json#L33